packages/adapter-components/src/definitions/system/deploy/deploy.ts

```diff
@@ -103,6 +111,8 @@ export type InstanceDeployApiDefinitions<AdditionalAction extends string, Client
   // fail if the change already has (error-level) errors from previous actions
   // default: true
   failIfChangeHasErrors?: boolean
+
+  recurseIntoTypes?: Record<string, RecurseIntoTypeDef>
```

Comment:
* not sure I understand what it means that the key is the type - what if there are multiple fields with the same type (think of a linked list with `before` and `after`)? I also don't know that we want to assume all the fields of the same nested type would have identical `recurseInto` definitions
* I think we should place this higher up (right after/before `requestsByAction`) with details explaining how this impacts what goes into which deploy definition and what's expected from the developer?

Reply:
Good point, I didn't think of the case when a few fields might have the same type. 1
So the alternative IMO is making `recurseIntoTypes` an array
I think the key shouldn't be the field (as in `standalone` definition), cause there might be scenarios when the field is nested

so something like 
```suggestion
  recurseIntoTypes?: RecurseIntoTypeDef[]
  
  
export type RecurseIntoTypeDef = {
  typeName: string
  fieldPath: string[]
  changeIdFields: string[]
  condition?: DeployRequestCondition
  // when missing, all actions are used. currently only standard actions are supported.
  onActions?: ActionName[]
}
  
```

WDYT @netama ?

Reply:
we can start with that (should maybe rename `recurseIntoTypes`to just `recurseInto` with some explanation?), might want to revisit later but it shouldn't block us for now. 
does it have an effect on the "main" change or are users expected to omit the fields in the main deploy?

I think we should still make the `fieldPath` the more "central" thing here - this is about breaking down the parent element, and we're pointing to the child type that contains the actual definition.